packages/protocol-kit/src/SafeFactory.ts

```diff
@@ -62,7 +63,38 @@ class SafeFactory {
   }: SafeFactoryInitConfig) {
     this.#provider = provider
     this.#signer = signer
-    this.#safeProvider = new SafeProvider({ provider, signer })
+    const isPasskeySigner = signer && typeof signer !== 'string'
```

Comment:
Consider on extracting this to a local util since its reused.

Reply:
done! I addressed this comment in [this other PR](https://github.com/safe-global/safe-core-sdk/pull/841/). I created a util in called [`createSafeProvider`](https://github.com/safe-global/safe-core-sdk/pull/841/files#diff-6800aef51d104476f1848ec54d4786a4ab2548533edd3ea5c0c584fbfbf032edR16) to reuse all the logic behind this:

![Captura de pantalla 2024-06-04 a las 19 20 29](https://github.com/safe-global/safe-core-sdk/assets/26763673/05815283-05fe-405a-96bc-d4118b3a8ffc)

Reply:
After your suggestion[ in this comment](https://github.com/safe-global/safe-core-sdk/pull/841#discussion_r1627310306), I create an `static async init` method in the `SafeProvider` instead of the [`createSafeProvider`](https://github.com/safe-global/safe-core-sdk/pull/841/files#diff-6800aef51d104476f1848ec54d4786a4ab2548533edd3ea5c0c584fbfbf032edR16)